src/lib/util/heap.c

```diff
@@ -214,30 +214,22 @@ int fr_heap_extract(fr_heap_t *hp, void *data)
 	int32_t parent, child, max;
 
 	/*
-	 *	Extract element.  Default is the first one (pop)
+	 *	Extract element.
 	 */
-	if (!data) {
-		if (unlikely((hp->num_elements == 0) || !hp->p[0])) {
-			fr_strerror_const("Tried to extract element from empty heap");
-			return -1;
-		}
-		parent = 0;
-	} else {		/* extract from the middle */
-		parent = index_get(hp, data);
+	parent = index_get(hp, data);
```

Comment:
`index_get()` dereferences `data`.  For safety, `fr_heap_extract()` should check for `data==NULL`, and return an error if that's passed to it.